apps/backend/src/core/logging.py

```
@@ -0,0 +1,87 @@
+"""
+Structured logging configuration with request-scoped correlation IDs.
+
+- Provides setup_logging() to configure structlog + stdlib bridge
+- Exposes request_id_var ContextVar and helpers
+"""
+from __future__ import annotations
+
+from contextvars import ContextVar
+from datetime import datetime, timezone
+import logging
+from typing import Any, MutableMapping
```

Ruff (0.12.2)

Import from `collections.abc` instead: `MutableMapping`

Import from `collections.abc`

(UP035)